src/channels.rs

```diff
@@ -24,20 +25,33 @@ impl MiEI {
         Ok(())
     }
 
-    pub fn get_role_id(&self, role_name: &str) -> Option<RoleId> {
-        self.courses
-            .values()
-            .filter_map(|x| x.courses.get(role_name))
-            .next()
-            .map(|x| x.role.parse::<RoleId>().unwrap())
+    pub fn get_role_id(&self, role_name: &str) -> Vec<RoleId> {
+        let years = &self.courses;
+        let regex = Regex::new("([0-9]+)ANO([0-9]+)(SEMESTRE|SEM)").unwrap();
+        let year_regex = Regex::new("([0-9])+ANO").unwrap();
+        if regex.is_match(role_name) {
```

Comment:
`Regex::captures` returns an option. If you use `if let` you don't need to parse `role_name` twice and you don't need to `unwrap`
```
if let Some(splits) = regex.captures(role_name) {
    // use `splits`
}
```